util/start_test

```diff
@@ -1424,7 +1443,6 @@ def preprocess_options():
             escape_next_argument(sys.argv, index)
```

Comment:
Another option, instead of escaping/unescaping, would be to do your own arg parsing here -- pull ``--compopts --foo`` and ``--compopts --bar`` out of sys.argv, and keep ``compopts = [ "--foo", "--bar" ]`` off to the side to combine with the results of any ``--compopts=--baz`` the regular argparse handles.


Escaping/unescaping is a bit confusing, and means if someone did do the correct escaping to get the string ``--compopts='--foo'`` passed into start_test, you'll end up un-escaping the the string the user "meant" to provide.  (Though this is silly.)


An aside: what's the ``arg.replace()`` doing here?  It's worried about a string that's e.g. ``--compopts`` but also contains quotes?

Reply:
We could, though I'm not sure that will be any more robust (or less prone to issues) than our current mechanism (and this has been an issue for years without anyone complaining too loudly) 

The downside to doing our own parsing is that we lose all the functionality that argparse already provides (--help output, tab-completion support, error handling, and other things). And note that we can't just pull out `--compopts`/`--execopts`, we also have to pull out the next arg and handle the `--compopts=` case. 

We will currently pull out quotes that we didn't add for something like `--compopts='--fast'`, but at least for the cases that come to mind, that's not an issue. i.e. `--compopts='--fast --fast'` will still work since argparse has already parsed it and we're interpreting it as a single arg. If there are cases where that can hurt us, maybe we could build up a dict of which args we changed instead of always removing ones that start with quotes or something.

`arg.replace()` comes from https://github.com/chapel-lang/chapel/pull/2374. I don't remember the exact details off the top of my head, but without that change `$CHPL_HOME/util/paratest.server -compopts <option>` had `<option>` getting truncated (and note we're not replacing the value in sys.argv, just a local temp copy)

Reply:
Hmm.  I can't figure out how removing quotes from the ``arg`` there (that should correspond to the word ``--compopts``, not the comp opt being passed) could have helped with that.  Though paratest.server is a bit complicated, and adds lots of quotes to things.

>  i.e. something like '-compopts --no-checks' would be seen by start_test as "--no-check"

I wonder if the "something like" was actually "-no-check", because it was hitting the unescaping-something-that's-not-escaped bug you're fixing here.

Reply:
I did mean just those four things and their next args, leaving the rest of the command line for the usual argparse handling.

I don't think you'd have to handle the ``--compopts=--baz`` case, since argparse will handle that without any escaping needed.  You would need to combine the opts you collected with the ones argparse collected though.

It wouldn't handle ``--compopts= --foo`` (with a space after the ``=``) (If that's supposed to be allowed), but neither will the current code or the code after this PR.


Testing that was a little confusing, because one of the first tests it tries to run is ``Samples/Performance/foo``.  But that's unrelated to the ``--foo``.  :laughing:

Reply:
Though the separate set of compopts "off to the side" would mean losing the order between ``--compopts --foo --compopts=--bar --compopts --baz``

Reply:
Oh well.  I think the case I'm talking about -- where you'd still end up unescaping something you didn't escape to start with, because it already has ``'``s -- is really silly and not worth worrying about anyway.

Reply:
> Hmm. I can't figure out how removing quotes from the arg there (that should correspond to the word --compopts, not the comp opt being passed) could have helped with that. Though paratest.server is a bit complicated, and adds lots of quotes to things.

>>i.e. something like '-compopts --no-checks' would be seen by start_test as "--no-check"

> I wonder if the "something like" was actually "-no-check", because it was hitting the unescaping-something-that's-not-escaped bug you're fixing here.

Yea.. I can't quite figure it out either, but I just used the "not sure off the top of my head" excuse ;) Probably worth digging into a little more, but I'll punt until post-release when I'm thinking about merging this to dig any further.

Reply:
And if you can think of a compelling case where our escaping/unescaping really breaks things I'd be interested, but beyond that I really, really don't want to do custom argument processing. argparse offers a ton of functionality, and I don't want to try to replicate any of that even if it's just for 2 options.

Reply:
Nah, I just saw a loose thread and was pulling on it.  I agree that nothing it's led to is worth pursuing.